src/main/java/apoc/util/FileUtils.java

```diff
@@ -42,68 +42,71 @@ public static CountingInputStream inputStreamFor(String fileName) throws IOExcep
+        try {
+            StreamConnection streamConnection = HDFSUtils.readFile(fileName);
+            Reader reader = new BufferedReader(new InputStreamReader(streamConnection.getInputStream(), "UTF-8"));
+            return new CountingReader(reader, streamConnection.getLength());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static CountingReader readFile(String fileName) throws IOException, FileNotFoundException {
+        File file = new File(fileName);
         if (!file.exists() || !file.isFile() || !file.canRead()) throw new IOException("Cannot open file "+fileName+" for reading.");
         return new CountingReader(file);
-	}
+    }
 
-	private static CountingInputStream readFileStream(String fileName) throws IOException, FileNotFoundException {
-		File file = new File(fileName);
+    private static CountingInputStream readFileStream(String fileName) throws IOException, FileNotFoundException {
+        File file = new File(fileName);
         if (!file.exists() || !file.isFile() || !file.canRead()) throw new IOException("Cannot open file "+fileName+" for reading.");
         return new CountingInputStream(file);
-	}
+    }
 
     public static String changeFileUrlIfImportDirectoryConstrained(String url) throws IOException {
         if (isFile(url) && ApocConfiguration.isEnabled("import.file.use_neo4j_config")) {
             if (!ApocConfiguration.isEnabled("import.file.allow_read_from_filesystem"))
                 throw new RuntimeException("Import file "+url+" not enabled, please set dbms.security.allow_csv_import_from_file_urls=true in your neo4j.conf");
+
             String importDir = ApocConfiguration.get("import.file.directory", null);
+
+            URI uri = URI.create(url);
+            if(uri == null) throw new RuntimeException("Path not valid!");
+
             if (importDir != null && !importDir.isEmpty()) {
                 try {
-                    File dir = new File(importDir);
-                    if (!dir.exists() || !dir.isDirectory() || !dir.canRead()) throw new Exception();
-                    String fileAsUrl = null;
-                    try { fileAsUrl = new URI(url).getPath(); } catch (Exception e) { /*no protocol*/ }
-                    if ("".equals(fileAsUrl)){
-                        String newUrl = url.replace("file://", "file:///");
-                        fileAsUrl = new URI(newUrl).getPath();
-                        if ("".equals(fileAsUrl)) fileAsUrl=newUrl.replace("file://","");
-                    }
-                    File file = url.startsWith(importDir) ? new File(fileAsUrl) : new File(dir.getAbsolutePath(), fileAsUrl);
-                    if (!file.exists() || !file.isFile() || !file.canRead()) throw new Exception();
-                    return new URL("file","",file.getAbsolutePath()).toString();
+                    String relativeFilePath = !uri.getPath().isEmpty() ? uri.getPath() : uri.getHost();
```

Comment:
where did all that code go?

Reply:
As we moved to `URI#create method this allowed us to simplify the code base